lewis/examples/modbus_device/__init__.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# lewis - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.devices import Device
+from lewis.adapters.modbus import ModbusAdapter, ModbusDataBank
+
+
+class ModbusDevice(Device):
+    pass
+
+
+class ModbusInterface(ModbusAdapter):
+    """
+    The class attributes di, co, ir and hr represent Discrete Inputs, Coils, Input Registers and
+    Holding Registers, respectively. Each attribute should be assigned a ModbusDataBank instance
+    by the Interface implementation.
+
+    Here, two basic ModbusDataBanks are created and initialized to a default value across the full
+    range of valid addresses. One DataBank is shared by di and co, and the other by ir and hr to
+    demonstrate overlaid memory segments. If you want each segment to have its own memory, just
+    create separate instances for all four.
+    """
+    di = ModbusDataBank.create_basic(False)
+    co = di
+    ir = ModbusDataBank.create_basic(0)
+    hr = ir
```

Comment:
Yes, it's weird that the `Device` is just a `pass`. But that's just a consequence of not having bindings yet. We can just update the example once we implement those.

> Could `di`, `co`, `ir`, `hr` exist in the `Device` as well?

I had a similar thought at some point, but dismissed it because this would break our model of system separation, no? `Device` is supposed to be just a Python class that knows nothing of protocol or adapters, and `Interface` acts as the binding between `Adapter` and `Device`?